ROX-18132: compliance scan setting proto and stores

dashrews78 opened the pull request:
## Description

Proto and stores for compliance operator scan schedules

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

dashrews78 committed:
X-Smart-Branch-Parent: master

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -4,9 +4,17 @@ option go_package = "storage";
 package storage;
 
+enum NodeType {
+    INFRA = 0;
```
After adding the above comment, I started wondering what other name would communicate better.

dashrews78 replied:
There is an INFRA node role in openshift.  https://access.redhat.com/solutions/5034771.  I should probably re-name this to NodeRole as that is what I was getting at.

md2119 replied:
Let's rename it. Reading the object, I thought the purpose was to add a reusable NodeType type.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -41,3 +49,31 @@ message ComplianceOperatorRuleV2 {
+        string                    profile_id               = 1 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorProfileV2:id),no-fk-constraint"'];
+    }
+    repeated Profile              profiles                 = 7;
+    message Cluster {
+        string                    cluster_id               = 1 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"fk(Cluster:id),no-fk-constraint,type(uuid)"'];
+        string                    namespace                = 2 [(gogoproto.moretags) = 'search:"Namespace,store"'];
+        string                    namespace_id             = 3 [(gogoproto.moretags) = 'search:"Namespace ID" sql:"fk(NamespaceMetadata:id),no-fk-constraint,type(uuid)"'];
+    }
```
Namespace is not a property of cluster. This structure seems a bit absurd. Could you elaborate on structure?

dashrews78 replied:
yeah left over artifact from a cluster UT that started failing because I didn't have the search category set in scan settings `gen.go`.  Will remove the namespace stuff.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -41,3 +49,31 @@ message ComplianceOperatorRuleV2 {
+    storage.Schedule              schedule                 = 11;
+    google.protobuf.Timestamp     created_time             = 12;
+    google.protobuf.Timestamp     last_updated_time        = 13;
+    google.protobuf.Timestamp     last_executed_time       = 14;
```
1. I want to make sure that the status is captured somewhere. 
2. We keep this table focused on user-configured scan config objects and the last_executed_time can be captured in the status table.

dashrews78 replied:
@md2119 Status as far as the settings is concerned is whether or not we were successful applying it from sensor, which in theory could be different per cluster.  Correct?

Noted on last_executed_time.  I'm working on the scans/results now and saw that that nugget of info makes more sense at that level.

md2119 replied:
> Status as far as the settings is concerned is whether or not we were successful applying it from sensor, which in theory could be different per cluster. Correct?

Correct. Error from the last compliance scan job would go where your capture a compliance job run which I think you hinted in your second point.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -41,3 +49,31 @@ message ComplianceOperatorRuleV2 {
+    google.protobuf.Timestamp     created_time             = 12;
+    google.protobuf.Timestamp     last_updated_time        = 13;
+    google.protobuf.Timestamp     last_executed_time       = 14;
+    SlimUser                      user                     = 15;
```
Is this a user who created or updated the settings?

dashrews78 replied:
yes.  having created and updated users seemed a bit much.  figured just one would be sufficient.

md2119 replied:
then my nit would be to rename the var to creator/created_by.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -41,3 +49,31 @@ message ComplianceOperatorRuleV2 {
+    }
+    repeated Cluster              clusters                 = 8;
+    repeated NodeType             node_roles               = 9;
+    bool                          strict_node_scan         = 10;
```
I think this should be ACS wide setting to keep this simple for users and get a uniform behavior across clusters.

md2119 replied:
If you are on the same page then I recommend adding the env var to https://github.com/stackrox/stackrox/blob/master/pkg/env/compliance_operator.go

dashrews78 committed:
X-Smart-Squash: Squashed 21 commits:

165cb511ff profile rules stores and protos
d0387b24f0 datastore
b1022cae93 some extra characters
92de045445 style
eeb586e99a terminate quote
be6c233d97 switch key
de25798169 style
0a057b01c1 adjust the repeated
4c81c7f075 fix test
2ae7a50a58 some rework
f9dc034c67 fix search field
51c8453d9a rebase to regenerate
48eac479b6 starting
cde690fd47 first draft
6c098d8772 regen
9788d825d2 fix gen file
ece6ee8216 fix generator
41b61c35d5 trying something
bf0e82b307 fix cluster test
d4fef06dde missed a file
7a5039c7a3 some rework

dashrews78 committed:
some additional comment related rework

dashrews78 commented:
/retest

dashrews78 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -41,3 +43,36 @@ message ComplianceOperatorRuleV2 {
+        string                    profile_id               = 1 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorProfileV2:id),no-fk-constraint"'];
+    }
+    repeated Profile              profiles                 = 7;
+    message ClusterScanStatus {
```
Felt like we should store errors per cluster.
